### PR TITLE
wait for tabs to load when reinjecting

### PR DESCRIPTION
**Pull request by tophf**
Didn't really test it, just wrote the fix based by the symptoms described in https://github.com/openstyles/stylus/pull/843#discussion_r377807448: the reinjector will wait for the tabs to finish loading before pinging them.

>When the browser starts, there is a chance that the pong callback/content script is not available yet but the background script started pinging. As a result, content scripts defined in manifest are injected and background also tries to inject content scripts again. Therefore the error occurs.

>I guess this can be produced consistently if there is a heavy processing content script (which can be injected by other extensions) is injected before `apply.js`.

### CODE REVIEW

**Comment by narcolepticinsomniac** [comment]
From the description, I figured this PR was attempting to get rid of those errors instead of suppressing them. Not really sure though, because that's not what I'm seeing. Reload still triggers `Error: Extension context invalidated.`, and the home page on launch has "already declared" syntax errors.

**Comment by eight04** [comment]
Instead of waiting for busy tabs, it should be easier to move [the injection](https://github.com/openstyles/stylus/blob/438fdebc5c7ebd2e7f728a7295f6db754d575b6a/background/background.js#L267) to the [onInstalled event](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/runtime/onInstalled). Ideally, we only have to inject scripts when the extension is installed/updated. Though I'm not sure if reloading an extension is counted as an update. Need investigation.

**Comment by tophf** [comment]
Ah, the repro steps evaded my attention, now it should be fixed.

As for "Extension context invalidated" it's not related to this problem and will be fixed by the style injector PR (#843). 

>move the injection to the onInstalled event

Nope, AFAIK it won't work after simply re-enabling of a disabled extension.

**Comment by eight04** [comment]
> Reload still triggers Error: Extension context invalidated.

This PR doesn't fix this error. See https://github.com/openstyles/stylus/issues/770

> the home page on launch has "already declared" syntax errors.

Could you try commenting out [these lines](https://github.com/openstyles/stylus/blob/438fdebc5c7ebd2e7f728a7295f6db754d575b6a/background/background.js#L266-L268) and see if the `SyntaxError` disappears?

**Comment by tophf** [comment]
@eight04 I've already fixed this, refresh the page :-)

File: background/content-scripts.js [hunk 1]
```diff
@@ -54,15 +54,29 @@ const contentScripts = (() => {
...
+          busyTabs.add(tab.id);
+        } else {
           injectToTab({
             url: tab.url,
             tabId: tab.id
           });
         }
       }
+      if (busyTabs.size) {
+        chrome.tabs.onUpdated.addListener(function _(tabId, {status}, {url}) {
```
Comment (eight04) [inline background/content-scripts.js]: Do we need to listen to `onRemoved` event?

Comment (tophf) [reply background/content-scripts.js]: Yeah, even though it's quite unlikely but it'd be more reliable.

**Comment by narcolepticinsomniac** [comment]
Syntax errors are fixed.

>As for "Extension context invalidated" it's not related to this problem and will be fixed by the style injector PR

My fault. I wasn't even sure the objective of this PR. I'm still not, tbh. Guess it's just a re-injection optimization?

**Comment by tophf** [comment]
Well the old behavior may be considered a bug, strictly speaking, so it's both an optimization and a fix.

**Pull request merged**